From c9ddcc6866f55956bf4b39715e237f45d1a533b3 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Mon, 15 Jul 2024 17:58:33 +0200
Subject: [PATCH] iptables: fix compat with legacy

Fix case where iptables binary uses nft backend in LXC container.

iptables-save may not output anything at all, while
/proc/net/ip_tables_names exists, so we check for both before concluding
that firewall is not enabled in kernel. More importantly, it prevents
awall activate --force to exit with failure.

fixes commit 49e6185321fc (iptables: fix to work with nft backend)
---
 awall/iptables.lua | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/awall/iptables.lua b/awall/iptables.lua
index 754f076..d41553b 100644
--- a/awall/iptables.lua
+++ b/awall/iptables.lua
@@ -23,10 +23,10 @@ local M = {}
 
 local families = {
 	inet={
-		cmd='iptables', file='rules-save'
+		cmd='iptables', file='rules-save', procfile='/proc/net/ip_tables_names'
 	},
 	inet6={
-		cmd='ip6tables', file='rules6-save'
+		cmd='ip6tables', file='rules6-save', procfile='/proc/net/ip6_tables_names'
 	}
 }
 
@@ -64,7 +64,7 @@ local function actfamilies()
 	if _actfamilies then return _actfamilies end
 	_actfamilies = {}
 	for _, family in ipairs(ACTIVE) do
-		if #acttables(family) > 0 then
+		if posix.stat(families[family].procfile) or #acttables(family) > 0 then
 			table.insert(_actfamilies, family)
 		else printmsg('Warning: firewall not enabled for '..family) end
 	end
-- 
2.45.2

